yellowbrick/features/scatter.py

```diff
@@ -0,0 +1,244 @@
+            names are selected as the columns of the DataFrame.
+
+        :param classes: a list of class names for the legend
+            If classes is None and a y value is passed to fit then the classes
+            are selected from the target vector.
+
+        :param color: optional list or tuple of colors to colorize lines
+            Use either color to colorize the lines on a per class basis or
+            colormap to color them on a continuous scale.
+
+        :param colormap: optional string or matplotlib cmap to colorize lines
+            Use either color to colorize the lines on a per class basis or
+            colormap to color them on a continuous scale.
+
+        :param kwargs: keyword arguments passed to the super class.
+
+        These parameters can be influenced later on in the visualization
+        process, but can and should be set as early as possible.
+        """
+        super(ScatterVisualizer, self).__init__(
+            ax, features, classes, color, colormap, **kwargs
+        )
+        # Ensure with init that features doesn't have more than two features
+        if features is not None:
+            if len(features) != 2:
+                raise YellowbrickValueError('ScatterVisualizer only accepts two features')
+
+    def fit(self, X, y=None, **kwargs):
+        """
+        The fit method is the primary drawing input for the parallel coords
+        visualization since it has both the X and y data required for the
+        viz and the transform method does not.
+
+        Parameters
+        ----------
+        X : ndarray or DataFrame of shape n x m
+            A matrix of n instances with m features
+
+        y : ndarray or Series of length n
+            An array or series of target or class values
+
+        kwargs : dict
+            Pass generic arguments to the drawing method
+
+        Returns
+        -------
+        self : instance
+            Returns the instance of the transformer/visualizer
+        """
+        nrows, ncols = X.shape
+
+        # Store the classes for the legend if they're None.
+        if self.classes_ is None:
+            # TODO: Is this the most efficient method?
+            self.classes_ = [str(label) for label in set(y)]
+
+        # Handle the feature names if they're None.
+        if self.features_ is None:
+                # If X is a data frame, get the columns off it.
+                if is_dataframe(X) and X.columns == 2:
+                    self.features_ = X.columns
+
+                # Otherwise create numeric labels for each column.
+                elif ncols == 2:
+                    self.features_ = [
+                        str(cdx) for cdx in range(ncols)
+                    ]
+                else:
+                    raise YellowbrickValueError('ScatterVisualizer only accepts two features, please explicted set the features key argument or only pass a matrix/ dataframe with only two columns')
```

Comment:
Your error message says " please [explicitly] set the features key argument or only pass a matrix/ dataframe with only two columns...." In combo with row 162, this seems to imply you meant for the 'features' argument to be able to slice you dataframe using column names? But, this doesn't happen anywhere. It looks like you need to add the `else` statement to correspond to row 162 that decides how to use `features_` to slice the `X` dataframe to match your `__init__` docstring. In addition, change your `fit` docstring describing what X is. 

OR, abandon this bonus functionality and require an m-width X and edit all documentation to match. It's a nice-to-have feature but seems like it might be something we want to only implement if it is done consistently across other feature visualizers.